jax/experimental/fused_attention_stableHLO_test.py

```diff
@@ -0,0 +1,124 @@
+import argparse
+from functools import partial
+from typing import Any, Optional
+from flax import linen as nn
+import jax
+import jax.numpy as jnp
+from jax import lax
+from jax.experimental.fused_attention_stableHLO import dot_product_attention
+
+from jax import core, vmap
+import os
+
+Array = jnp.ndarray
+DType = jnp.dtype
+PRNGKey = jnp.ndarray
+
+
+def f(input_q: Array,
+    input_k: Array,
+    input_v: Array,
+    bias: Optional[Array] = None,
+    mask: Optional[Array] = None,
+    mask_type: Optional[str] = None,
+    scale: float = 0.5) -> Array:
+
+    output = dot_product_attention(
+        input_q,
+        input_k,
+        input_v,
+        scale=scale,
+        bias=bias,
+        mask=mask,
+        is_cauasl_mask=False,
+        dropout_rate=0.1)
+
+    return output
+
+def train_step(input_q, input_k, input_v, bias, mask, grad):
+    out, f_vjp = jax.vjp(
+        vmap(partial(f, scale=4.0, mask_type=""), in_axes=(0, 0, 0, 0, 0)),
+        input_q, input_k, input_v, bias, mask
+    )
+    input_q_grad, input_k_grad, input_v_grad, bias_grad, _ = f_vjp(grad)
+    return out, (input_q_grad, input_k_grad, input_v_grad, bias_grad, _)
+
+def main():
+    parser = argparse.ArgumentParser(description='T5X MHA Unit Test')
+    parser.add_argument("--batch_size", dest="batch_size", type=int, default=16)
+    parser.add_argument("--q_seq_len", dest="q_seq_len", type=int, default=256)
+    parser.add_argument("--kv_seq_len", dest="kv_seq_len", type=int, default=256)
+    parser.add_argument("--num_attn_heads", dest="num_attn_heads", type=int, default=16)
+    parser.add_argument("--head_dim", dest="head_dim", type=int, default=64)
+    parser.add_argument("--scale", dest="scale", type=float, default=1.0)
+    parser.add_argument("--dropout_rate", dest="dropout_rate", type=float, default=0.)
+    parser.add_argument("--bias", action="store_true")
+    parser.add_argument("--mask", action="store_true")
+    parser.add_argument("--mask_type", dest="mask_type", type=str, default=None)
+    parser.add_argument("--fwd_only", dest='fwd_only', action="store_true")
+    parser.add_argument("--xla_dump_to", dest="xla_dump_to", type=str, default=None)
+    args = parser.parse_args()
+   
+    # Set XLA flags with additional flags when xla_dump_to is provided
+    xla_flags = '--xla_gpu_enable_triton_gemm=false --xla_gpu_simplify_all_fp_conversions --xla_gpu_fused_attention_use_cudnn_rng=true --xla_gpu_enable_cudnn_fmha=true --xla_gpu_enable_xla_runtime_executable=false --xla_gpu_fused_attention_use_cudnn_rng=true'
```

Comment:
Can you remove the first 2 flags?
xla_gpu_fused_attention_use_cudnn_rng is duplicated.
Can you also remove xla_gpu_enable_xla_runtime_executable?
Document why all the non FA flags that you use are set.

Most customers won't set those flags by default while using dot_product_attention(). So we need to document/handle correctly that case.

Reply:
removed.